tests/validation/tests/v3_api/test_bkp_restore_s3_recover.py

```diff
@@ -0,0 +1,158 @@
+        "folder": AWS_S3_BUCKET_FOLDER_NAME,
+        "region": AWS_REGION,
+        "endpoint": "s3.amazonaws.com"
+    }
+}
+
+
+@if_test_all_snapshot
+def test_bkp_restore_s3_recover_validate():
+    cluster = namespace["cluster"]
+    client = namespace["client"]
+    ns, b_info = validate_backup_create(namespace, backup_info, "s3")
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) > 0
+    ips_to_remove = []
+    for etcd_node in etcd_nodes:
+        ips_to_remove.append(etcd_node.externalIpAddress)
+        client.delete(etcd_node)
+    cluster = wait_until_cluster_is_updating(client, cluster)
+    for ip_to_remove in ips_to_remove:
+        delete_node_from_ec2(ip_to_remove)
+    ips_to_remove.clear()
+    cluster = add_new_etcd_nodes(client, cluster)
+    cluster = client.reload(cluster)
+    message = "Please restore your cluster from backup"
+    cluster = wait_for_cluster_transitioning_message(client, cluster, message)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    assert len(etcd_nodes) == 3
+    restore_backup_to_recover(client, cluster, b_info)
+    etcd_nodes = get_etcd_nodes(client, cluster)
+    master_node = get_worker_nodes(client, cluster)
+    validate_ingress(namespace["p_client"], cluster,
+                     [b_info["workload"]], ns["host"],
+                     "/name.html")
+    if RANCHER_CLEANUP_CLUSTER:
+        for node in etcd_nodes+master_node:
+            ips_to_remove.append(node.externalIpAddress)
+            client.delete(node)
+        for ip_to_remove in ips_to_remove:
+            delete_node_from_ec2(ip_to_remove)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def create_project_client_and_cluster_s3_three_etcd(node_template_ec2,
+                                                    request):
+    node_roles = [
+        ["controlplane", "worker"],
+        ["etcd"], ["etcd"], ["etcd"],
+    ]
+    cluster, aws_nodes = create_and_validate_custom_host(node_roles,
+                                                         rke_config=rke_config)
+    client = get_user_client()
+    cluster = get_cluster_by_name(client, CLUSTER_NAME)
+
+    p, ns = create_project_and_ns(USER_TOKEN, cluster, "testrecover")
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    c_client = get_cluster_client_for_token(cluster, USER_TOKEN)
+
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+    namespace["c_client"] = c_client
+    namespace["client"] = client
+
+    def fin():
+        cluster_cleanup(client, cluster)
+    request.addfinalizer(fin)
+
+
+def wait_for_cluster_transitioning_message(client, cluster, message):
+    start = time.time()
+    while message not in cluster.transitioningMessage:
+        print(cluster.transitioningMessage)
+        time.sleep(5)
+        cluster = client.reload(cluster)
+        if time.time() - start > MACHINE_TIMEOUT:
```

Comment:
In Worst case scenario, we wait for `MACHINE_TIMEOUT` which is 1200 seconds here? Is that needed? How long does it usually take for the message to appear?

Reply:
I will use `DEFAULT_TIMEOUT`